packages/graphql-api-commons/src/providers/LoggerProvider.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import pino, { Logger } from 'pino'
+import { inject, singleton } from 'tsyringe'
+import { IBaseConfiguration } from '../configuration'
+import { BaseInjectorNames } from '../injectors'
+import { Provider } from './Provider'
+
+@singleton()
+export class LoggerProvider<TConfiguration extends IBaseConfiguration>
+	implements Provider<Logger>
+{
+	private _logger: Logger
+
+	public constructor(
+		@inject(BaseInjectorNames.Configuration) private _config: TConfiguration,
+	) {
+		this._logger = pino({
+			serializers: {
+				err: pino.stdSerializers.err,
+				error: pino.stdSerializers.err,
+			},
+			prettyPrint: this._config.loggingPretty
+				? {
+						levelFirst: true,
+				  }
+				: undefined,
+			prettifier: this._config.loggingPretty
```

Comment:
Running this with `loggingPretty` set might lead to a runtime error. Looks like the latest versions of `pino` do not need this to work. So maybe this could be:

```typescript
this._logger = pino({
			serializers: {
				err: pino.stdSerializers.err,
				error: pino.stdSerializers.err,
			},
			prettyPrint: this._config.loggingPretty
				? {
						levelFirst: true,
				  }
				: undefined,
		})
```

More at: https://github.com/pinojs/pino-pretty/blob/master/Readme.md#programmatic-integration